corgidrp/detector.py

```diff
@@ -274,6 +280,69 @@ def slice_section(frame, obstype, key, detector_regions=None):
         raise Exception('Corners invalid. Tried to slice shape of {0} from {1} to {2} rows and {3} columns'.format(frame.shape, r0c0, rows, cols))
     return section
 
+
+def imaging_area_geometry(obstype='SCI',detector_regions=None):
+    """
+    
+    Return geometry of imaging area in reference to full frame.
+    
+    Ported from II&T
+
+    Args: 
+        obstype (str): Either 'SCI' or 'ENG'
+        detector_regions (dict): a dictionary of detector geometry properties. 
+
+    Returns: 
+        rows_im (int): Number of rows of imaging area
+        cols_im (int): Number of columns of imaging area
+        r0c0_im (tuple): Tuple of (row position, column position) of corner closest to (0,0)
+
+    """
+
+    if detector_regions is None:
+            detector_regions = detector_areas
+
+    _, cols_pre, _ = unpack_geom(obstype,'prescan', detector_regions = detector_regions)
+    _, cols_serial_ovr, _ = unpack_geom(obstype,'serial_overscan', detector_regions = detector_regions)
+    rows_parallel_ovr, _, _ = unpack_geom(obstype,'parallel_overscan', detector_regions = detector_regions)
+    fluxmap_rows, _, r0c0_image = unpack_geom(obstype,'image', detector_regions = detector_regions)
+
+    rows_im = detector_areas[obstype]['frame_rows'] - rows_parallel_ovr
+    cols_im = detector_areas[obstype]['frame_cols'] - cols_pre - cols_serial_ovr
+    r0c0_im = r0c0_image.copy()
+    r0c0_im[0] = r0c0_im[0] - (rows_im - fluxmap_rows)
+
+    return rows_im, cols_im, r0c0_im
+
+
+def imaging_slice(frame,obstype='SCI',detector_regions=None): 
```

Comment:
This is the same as `imaging_slice` further below in this file (which is the one your script must be calling since this comes before it).